app/services/ui_notifications/rss_notifications_checker.rb

```diff
@@ -0,0 +1,74 @@
+require 'rss'
+require 'date'
+
+module UINotifications
+  class RssNotificationsChecker
+    class Item
+      def initialize(feed_item)
+        @item = feed_item
+      end
+
+      def link
+        @link ||= @item.link.respond_to?(:href) ? @item.link.href : @item.link
+      end
+
+      def summary
+        @summary ||= begin
+          summary = @item.summary if @item.respond_to?(:summary)
+          summary = @item.description if @item.respond_to?(:description)
+          summary.respond_to?(:content) ? summary.content : summary
+        end
+      end
+
+      def title
+        @title ||= @item.title.respond_to?(:content) ? @item.title.content : @item.title
+      end
+    end
+
+    def initialize(options = [])
+      @url = options[:url] || Setting[:rss_url]
+      @latest_posts = options[:latest_posts] || 3
+      @force_repost = options[:force_repost] || false
+      @audience = options[:audience] || Notification::AUDIENCE_GLOBAL
+    end
+
+    def deliver!
+      # This is a noop every time rss_enable=false, the moment it
+      # gets enabled, notifications for RSS feeds are created again
+      return true unless Setting[:rss_enable]
+      feed = RSS::Parser.parse(@url, false)
+      feed.items[0, @latest_posts].each do |feed_item|
+        item = Item.new(feed_item)
+        blueprint = rss_notification_blueprint
+        if notification_already_exists?(item)
+          next unless @force_repost
+        end
+        Notification.create(
```

Comment:
I would create! so that exception is handled / logged somewhere.

Reply:
https://github.com/theforeman/foreman/pull/4240/files#diff-bbe58f068bd900b15ec6debded3fbcf8R8 - the `rescue_from` in the job logs any possible error from here

Reply:
@dLobatog the thing is that create fails silently returning false, not raising an exception, with `create!` we'd be able to find in logs that it failed and why